src/PriorityReliableQueue.php

```diff
@@ -0,0 +1,92 @@
+<?php
+namespace Altmetric;
+
+use Redis;
+use Psr\Log\LoggerInterface;
+
+class PriorityReliableQueue implements \Iterator
+{
+    public $name;
+    public $queues;
+    private $redis;
+    private $logger;
+    private $value;
+    private $queue;
+    private $workingQueue;
+
+    public function __construct($name, array $queues, Redis $redis, LoggerInterface $logger)
+    {
+        $this->name = $name;
+        $this->redis = $redis;
+        $this->logger = $logger;
+
+        foreach ($queues as $queue) {
+            $this->queues[$queue] = "{$queue}.working_on.{$name}";
+        }
+    }
+
+    public function rewind()
```

Comment:
I guess I'm just not used to `rewind` being stateful, but maybe that's okay for practical reasons 🤔